spec/requests/lead_providers/partnerships_spec.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "Lead provider partnerships spec", type: :request do
+  let(:user) { create :user, :lead_provider }
+  let!(:cohort) { create :cohort }
+
+  before do
+    sign_in user
+  end
+
+  describe "GET /lead-providers/partnerships/:id" do
+    let(:partnership) { create :partnership, cohort: cohort }
+
+    context "when the user is the lead provider for the school" do
+      it "should show the partnership page" do
+        get lead_providers_partnership_path(partnership)
+
+        expect(response).to render_template :show
+        expect(assigns(:partnership)).to eq partnership
+      end
+    end
+  end
```

Comment:
What about the context from the test file that this replaces,
```
    context "when school is not in a partnership with the lead provider" do
      it "should return not found" do
        expect {
          get lead_providers_school_detail_path(school)
        }.to raise_error(ActiveRecord::RecordNotFound)
      end
    end
```
Is this never going to happen by default, because partnership always has a school and a lead provider? What if a lead provider tries to access someone else's partnership?

Reply:
Yup, that's a  bit odd - to rewrite the test I'd need to test that when partnership_id is completely random then it fails with 404, which is kind of given. That test was previously required as an extra authorisation layer which is no longer needed...

And now when I wrote that it just occurred to me that you're right - we need to test lead provider is unable to access other providers partnerships, not only when the partnership is missing. For now I'll just add the test, but it would be nice to have partnership policy here.